### DISCUSSION TITLE
Rethink `jj log` argument handling

### DISCUSSION THREAD

**Discussion by robinst** [jj-vcs/jj#5064] [Ideas]
I'm currently using both `git` and `jj` and one of the things that trips me up frequently is that `log` behaves differently. I type:

```
jj log branchname
```

And get no output for a minute (big monorepo), until I remember that `jj` needs a `-r` in there 😢.

I don't know how this could be solved, but wanted to raise it. Anyone else run into this?

**Comment by robinst** [comment]
Maybe one way to make this less problematic is if `jj log foo` checked if `foo` is the name of a branch (or if it's **not** a file in the current commit), and if so, print a message, e.g.:

```
Log for path foo (if you meant revision, use -r)
```

**Reply by martinvonz** [reply]
We do almost that already:
```
$ jj log branchname
Warning: The argument "branchname" is being interpreted as a path. To specify a revset, pass -r "branchname" instead.
```

But we don't check until after the command has finished without producing any output.

**Reply by robinst** [reply]
Oh I see, so it's only a problem with big repos then! For me it takes minutes for the command to finish (I've always aborted the command before, which is why I've never seen it). How about showing it earlier? It could be after checking the first commit, or N commits, or maybe a time (not sure how easy that would be to implement though).

**Reply by arxanas** [reply]
My argument has been that we should forbid flag-less path arguments here and elsewhere, and require `-p`/`--path` (or maybe `-f`/`--file`) as per https://github.com/jj-vcs/jj/issues/3809; @robinst I'm curious if that would be intuitive for you?

**Reply by robinst** [reply]
Yes, requiring `-f` or `-p` sounds good to me too!

Personally I use my IDE for file history. But when I'm looking at using `jj` to rebase or squash a branch I might want to see its log, so log with a branch name is a more frequent operation than log for a path for me.

**Comment by simonmichael** [comment]
Just for the record, yes I also trip up on the need for -r at least in first few weeks of jj use.
